### DISCUSSION TITLE
Scroll on anchor click

### DISCUSSION THREAD

**Discussion by VottusCode** [idiotWu/smooth-scrollbar#330] [Q&A] [answered]
## Issue Summary
I'm not sure whether this is implemented, but I recently tried this library in one of my projects and foudn out that clicking on anchor links with hashes (eg href=#about-us) does not scroll. I've solved this issue with the code below.

```ts
const bar = Scrollbar.init(document.querySelector('#my-scrollbar'));

bar.containerEl.querySelectorAll("a[href*='#']").forEach(el => {
  el.addEventListener("click", () => {
    bar.scrollIntoView(document.getElementById(el.getAttribute("href").substring(1)))
  })
})
```

## Expected Behavior
When clicked on such anchor tag, it should scroll to desired element.

## Online demo
https://codepan.egoist.sh/gist/5d0770947026e19d2e29585e70b57d4c

**Comment by dolphin-wood** [comment, marked as answer]
See https://github.com/idiotWu/smooth-scrollbar/issues/128#issuecomment-390980479
